File: repo/work/Desktop/work/Programming/Java/Vibe Coding/Calculator.java
import java.util.Scanner;

public class Calculator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Simple Java Calculator");
        System.out.print("Enter first number: ");
        double num1 = readDouble(scanner);

        System.out.print("Enter operator (+, -, *, /): ");
        String operator = scanner.next();

        System.out.print("Enter second number: ");
        double num2 = readDouble(scanner);

        double result;
        switch (operator) {
            case "+":
                result = num1 + num2;
                break;
            case "-":
                result = num1 - num2;
                break;
            case "*":
                result = num1 * num2;
                break;
            case "/":
                if (num2 == 0) {
                    System.out.println("Error: Division by zero is not allowed.");
                    scanner.close();
                    return;
                }
                result = num1 / num2;
                break;
            default:
                System.out.println("Error: Unsupported operator.");
                scanner.close();
                return;
        }

        System.out.printf("Result: %.4f %s %.4f = %.4f%n", num1, operator, num2, result);
        scanner.close();
    }

    private static double readDouble(Scanner scanner) {
        while (!scanner.hasNextDouble()) {
            System.out.print("Invalid input. Please enter a number: ");
            scanner.next();
        }
        return scanner.nextDouble();
    }
}

File: repo/work/Desktop/work/Programming/Java/Vibe Coding/calculator.py
def read_float(prompt):
    while True:
        try:
            return float(input(prompt))
        except ValueError:
            print("Invalid input. Please enter a valid number.")


def main():
    print("Simple Python Calculator")
    num1 = read_float("Enter first number: ")
    operator = input("Enter operator (+, -, *, /): ").strip()
    num2 = read_float("Enter second number: ")

    if operator == "+":
        result = num1 + num2
    elif operator == "-":
        result = num1 - num2
    elif operator == "*":
        result = num1 * num2
    elif operator == "/":
        if num2 == 0:
            print("Error: Division by zero is not allowed.")
            return
        result = num1 / num2
    else:
        print("Error: Unsupported operator.")
        return

    print(f"Result: {num1:.4f} {operator} {num2:.4f} = {result:.4f}")


if __name__ == "__main__":
    main()
